imap_processing/cli.py

```diff
@@ -104,37 +148,101 @@ def _validate_args(args):
     args : argparse.Namespace
         An object containing the parsed arguments and their values
     """
-    if args.instrument not in imap_processing.INSTRUMENTS:
+    if args.instrument not in imap_data_access.VALID_INSTRUMENTS:
         raise ValueError(
             f"{args.instrument} is not in the supported instrument list: "
             f"{imap_processing.INSTRUMENTS}"
         )
-    if args.level not in imap_processing.PROCESSING_LEVELS[args.instrument]:
+    if args.data_level not in imap_processing.PROCESSING_LEVELS[args.instrument]:
         raise ValueError(
-            f"{args.level} is not a supported data level for the {args.instrument}"
+            f"{args.data_level} is not a supported data level for the {args.instrument}"
             " instrument, valid levels are: "
             f"{imap_processing.PROCESSING_LEVELS[args.instrument]}"
         )
-    if args.data_dir:
-        data_path = Path(args.data_dir)
-        if not data_path.exists():
-            raise ValueError(f"Data directory {args.data_dir} does not exist")
-        # Set the data directory to the user-supplied value
-        imap_data_access.config["DATA_DIR"] = data_path
 
 
 class ProcessInstrument(ABC):
     """An abstract base class containing a method to process an instrument.
 
     Attributes
     ----------
-    level : str
+    data_level : str
         The data level to process (e.g. ``l1a``)
+    dependency_str : str
+        A string representation of the dependencies for the instrument in the
+        format: "[{
+            'instrument': 'mag',
+            'data_level': 'l0',
+            'descriptor': 'sci',
+            'version': 'v00-01',
+            'start_date': '20231212',
+            'end_date': '20231212'
+        }]"
+    start_date : str
+        The start date for the output data. Format: YYYYMMDD
+    end_date : str
+        The end date for the output data. Format: YYYYMMDD
+    version : str
+        The version of the data. Format: vxx-xx
+    upload_to_sdc : bool
+        A flag indicating whether to upload the output file to the SDC.
     """
 
-    def __init__(self, level, file_path):
-        self.level = level
-        self.file_path = file_path
+    def __init__(
+        self,
+        data_level: str,
+        dependency_str: str,
+        start_date: str,
+        end_date: str,
+        version: str,
+        upload_to_sdc: bool,
+    ) -> None:
+        self.data_level = data_level
+
+        # Convert string into a dictionary
+        self.dependencies = loads(dependency_str.replace("'", '"'))
```

Comment:
Are we dumping malformed json somewhere upstream? If so we should correct it there. Single quotes aren't allowed in the JSON spec, so I'm not sure we should be doing the replacement beforehand.

Reply:
I'm not sure, this piece is left over from the first draft (just moved). If we want to change the JSON that would be in batch_starter.

Reply:
@tech3371 where does that command get created? I think it just needs a `json.dumps()` surrounding the dependency argument which will invert that string escaping for us.

Reply:
here: https://github.com/IMAP-Science-Operations-Center/sds-data-manager/blob/dev/sds_data_manager/lambda_code/SDSCode/batch_starter.py#L315